src/events/getZNAs.ts

```diff
@@ -0,0 +1,15 @@
+import { ethers } from 'ethers';
+import { BigNumber } from '@ethersproject/bignumber';
+import { getActiveLinks } from './activeLinks';
+
+export async function getZNAs(contract: ethers.Contract) {
+    type Association = {
```

Comment:
Consider a higher level `types.ts` file for types like this and elsewhere so you can take advantaged of TypeScript more